test/octree/test_octree.cpp

```diff
@@ -1142,80 +1142,84 @@ TEST (PCL, Octree_Pointcloud_Nearest_K_Neighbour_Search)
   Indices k_indices_bruteforce;
   std::vector<float> k_sqr_distances_bruteforce;
 
-  for (unsigned int test_id = 0; test_id < test_runs; test_id++)
-  {
-    // define a random search point
+  for (const std::size_t maxObjsPerLeaf: {0, 5}) {
+    if (maxObjsPerLeaf != 0)
+      octree.enableDynamicDepth (maxObjsPerLeaf);
+    for (unsigned int test_id = 0; test_id < test_runs; test_id++)
+    {
+      // define a random search point
 
-    PointXYZ searchPoint (static_cast<float> (10.0 * rand () / RAND_MAX),
-                          static_cast<float> (10.0 * rand () / RAND_MAX),
-                          static_cast<float> (10.0 * rand () / RAND_MAX));
+      PointXYZ searchPoint (static_cast<float> (10.0 * rand () / RAND_MAX),
+                            static_cast<float> (10.0 * rand () / RAND_MAX),
+                            static_cast<float> (10.0 * rand () / RAND_MAX));
 
-    unsigned int K = 1 + rand () % 10;
+      unsigned int K = 1 + rand () % 10;
 
-    // generate point cloud
-    cloudIn->width = 1000;
-    cloudIn->height = 1;
-    cloudIn->points.resize (cloudIn->width * cloudIn->height);
-    for (std::size_t i = 0; i < 1000; i++)
-    {
-      (*cloudIn)[i] = PointXYZ (static_cast<float> (5.0  * rand () / RAND_MAX),
-                                     static_cast<float> (10.0 * rand () / RAND_MAX),
-                                     static_cast<float> (10.0 * rand () / RAND_MAX));
-    }
+      // generate point cloud
```

Comment:
Seems a lot of places, we could use CloudGenerator in this test file? Should we add it as a good first issue?

Reply:
Generally yes, but there are already several issues (and a whole "project") about better random generators in PCL.